google/cloud/aiplatform/training_jobs.py

```diff
@@ -2673,6 +2673,455 @@ def _model_upload_fail_string(self) -> str:
+                Required. The user-defined name of this TrainingPipeline.
+            optimization_objective (str):
+                Optional. Objective function the model is to be optimized towards.
+                The training process creates a Model that optimizes the value of the objective
+                function over the validation set. The supported optimization objectives:
+                "minimize-rmse" (default) - Minimize root-mean-squared error (RMSE).
+                "minimize-mae" - Minimize mean-absolute error (MAE).
+                "minimize-rmsle" - Minimize root-mean-squared log error (RMSLE).
+                "minimize-rmspe" - Minimize root-mean-squared percentage error (RMSPE).
+                "minimize-wape-mae" - Minimize the combination of weighted absolute percentage error (WAPE)
+                                      and mean-absolute-error (MAE).
+                "minimize-quantile-loss" - Minimize the quantile loss at the defined quantiles.
+            column_transformations (Optional[Union[Dict, List[Dict]]]):
+                Optional. Transformations to apply to the input columns (i.e. columns other
+                than the targetColumn). Each transformation may produce multiple
+                result values from the column's value, and all are used for training.
+                When creating transformation for BigQuery Struct column, the column
+                should be flattened using "." as the delimiter.
+                If an input column has no transformations on it, such a column is
+                ignored by the training, except for the targetColumn, which should have
+                no transformations defined on.
+            project (str):
+                Optional. Project to run training in. Overrides project set in aiplatform.init.
+            location (str):
+                Optional. Location to run training in. Overrides location set in aiplatform.init.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use to run call training service. Overrides
+                credentials set in aiplatform.init.
+        """
+        super().__init__(
+            display_name=display_name,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+        self._column_transformations = column_transformations
+        self._optimization_objective = optimization_objective
+
+    def run(
+        self,
+        dataset: datasets.Dataset,
+        target_column: str,
+        time_column: str,
+        time_series_identifier_column: str,
+        time_variant_past_only_columns: List[str],
+        time_variant_past_and_future_columns: List[str],
+        forecast_window_end: int,
+        period_unit: str,
+        period_count: int,
+        predefined_split_column_name: Optional[str] = None,
+        weight_column: Optional[str] = None,
+        static_columns: List[str] = [],
```

Comment:
We should avoid mutable default values. Perhaps set this to Optional and default to None.